wire-ios-data-model/Source/Model/User/ZMUser+LegalHoldRequest.swift

```diff
@@ -272,4 +276,41 @@ extension ZMUser: SelfLegalHoldSubject {
         needsToAcknowledgeLegalHoldStatus = false
     }
 
+    // MARK: - Fingerprint
+
+    public var fingerprint: String? {
+        guard let syncContext = managedObjectContext?.zm_sync,
+              let proteusProvider = syncContext.proteusProvider else {
+                  log.error("Could not access proteusProvider")
+                  return nil
+              }
```

Comment:
Is this safe though? It's a public method so it may be called from the UI. If `managedObjectContext` is the view context, then we must be on the main thread. I don't see anywhere we switch to the sync context. This may only show up as a runtime crash.

You may need to switch to the sync context thread then do the work you need to.